### PR TITLE
Added support for connections over IPv6.

### LINKED ISSUE #257: IPv6 support
I'm dealing with a production environment where memcached are IPv6 only. The code in _connect assumes IP connections are IPv4 "self.socket_module.AF_INET".

Do you have plans to handle "IPv6 URI", for example: ('inet6:memcache-a', 11211) ?

### PR DESCRIPTION
**Pull request by pope1ni**
Fixes #257.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -126,6 +126,7 @@ Credits
 * `Stephen Rosen <https://github.com/sirosen>`_
 * `Feras Alazzeh <https://github.com/FerasAlazzeh>`_
 * `Moisés Guimarães de Medeiros <https://github.com/moisesguimaraes>`_
+* `Nick Pope <https://github.com/pope1ni>`_
```
Comment (jparise) [inline README.rst:129]: 👍

File: pymemcache/client/base.py [hunk 2]
```diff
@@ -109,6 +109,27 @@ def check_key_helper(key, allow_unicode_keys, key_prefix=b''):
     return key
 
 
+def normalize_server_spec(server):
+    if isinstance(server, tuple) or server is None:
+        return server
+    if isinstance(server, list):
+        return tuple(server)  # Assume [host, port] provided.
+    if not isinstance(server, six.string_types):
+        raise ValueError('Unknown server specification provided.')
```
Comment (jparise) [inline pymemcache/client/base.py:118]: Maybe include the `server` value and/or type in the exception message?

**Review by jparise — COMMENTED** [review]
Do you think we should make any documentation updates to reflect IPv6 support or the multiple ways in which server specs can be provided?

File: pymemcache/client/base.py (hunk 2 above)
Comment (pope1ni) [reply pymemcache/client/base.py:118]: Sure. Will output the repr.

**Comment by pope1ni** [comment]
I've made a bunch of documentation changes to highlight different ways of providing the server parameter and highlighting that IPv6 is now supported.

**Review by jparise — APPROVED** [review]
Looks great!

**Pull request merged**